backend/lib/til/shareable_content.ex

```diff
@@ -80,6 +80,14 @@ defmodule Til.ShareableContent do
     |> Enum.map(&Post.populate_reaction_count/1)
   end
 
+  def get_paginated_posts(only_public, paging_params) do
```

Comment:
There is a lot responsibility in this function, consider pulling out small functions and then combine them in one place(this function is a good place for that I think)